vim-patch:8.1.{2185,2188,2190,2197}

janlazo opened the pull request.

janlazo committed:
vim-patch:8.1.2185: syntax test fails

Problem:    Syntax test fails.
Solution:   Add missing file patch.
https://github.com/vim/vim/commit/bbfd1562aeaa5b40b6451effc399846b692d6992

janlazo committed:
vim-patch:8.1.2188: build error for missing define

Problem:    Build error for missing define.
Solution:   Add missing change.
https://github.com/vim/vim/commit/2b78ab5d0c91c229715ae140a34978506343bde3

These "WILD_" macros are used in earlier vim patches.

blueyed commented:
FreeBSD was/is flaky, restarted Travis.

janlazo committed:
vim-patch:8.1.2190: syntax test fails on Mac

Problem:    Syntax test fails on Mac.
Solution:   Limit the window size to 20 rows.
https://github.com/vim/vim/commit/83e9a1ce75818a78c5ddf8dcfb820634ca6fabff

justinmk commented:
`ci/install.sh` failing, did treesitter upstream files change? are we not pinned to a specific treesitter version?

```
No such file or directory (os error 2)
The command "ci/install.sh" failed and exited with 1 during .
```

blueyed commented:
> `ci/install.sh` failing, did treesitter upstream files change? are we not pinned to a specific treesitter version?
> 
> ```
> No such file or directory (os error 2)
> The command "ci/install.sh" failed and exited with 1 during .
> ```

```
> tree-sitter-cli@0.15.11 install /home/travis/.nvm/versions/node/v8.12.0/lib/node_modules/tree-sitter-cli
> node install.js
Downloading https://github.com/tree-sitter/tree-sitter/releases/download/0.15.11/tree-sitter-linux-x64.gz
+ tree-sitter-cli@0.15.11
added 1 package from 1 contributor in 1.479s
/home/travis/build/neovim/neovim/node_modules/tree-sitter-cli -> /home/travis/.nvm/versions/node/v8.12.0/lib/node_modules/tree-sitter-cli
Install tree-sitter c parser
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  252k    0  252k    0     0   526k      0 --:--:-- --:--:-- --:--:--  526k
ambiguities:
  ✓ pointer declarations vs multiplications
  ✓ casts vs multiplications
  ✓ function-like type macros vs function calls
  ✓ function calls vs parenthesized declarators vs macro types
  ✓ Call expressions vs empty declarations w/ macros as types
  ✓ Comments after for loops with ambiguities
  ✓ Top-level macro invocations
types:
  ✓ Primitive types
  ✓ Type modifiers
crlf:
  ✓ Line comments with escaped CRLF line endings
preprocessor:
  ✓ Include directives
  ✓ Object-like macro definitions
  ✓ Function-like macro definitions
  ✓ Ifdefs
  ✓ General if blocks
  ✓ Preprocessor conditionals in functions
  ✓ Preprocessor conditionals in struct/union bodies
  ✓ Unknown preprocessor directives
declarations:
  ✓ Struct declarations
  ✓ Union declarations
  ✓ Enum declarations
  ✓ Struct declarations containing preprocessor directives
  ✓ Primitive-typed variable declarations
  ✓ Variable storage classes
  ✓ Composite-typed variable declarations
  ✓ Pointer variable declarations
  ✓ Typedefs
  ✓ Function declarations
  ✓ Function definitions
  ✓ Function specifiers after types
  ✓ Linkage specifications
  ✓ Type qualifiers
  ✓ Local array declarations
  ✓ Attributes
expressions:
  ✓ Number literals
  ��� Identifiers
  ✓ Common constants
  ✓ Function calls
  ✓ String literals
  ✓ Character literals
  ✓ Field access
  ✓ Boolean operators
  ✓ Math operators
  ✓ The comma operator
  ✓ Assignments
  ✓ Pointer operations
  ✓ Type-casts
  ✓ Sizeof expressions
  ✓ Compound literals
  ✓ Compound literals with trailing commas
  ✓ Comments with escaped newlines
  ✓ Comments with escaped chars and newlines
statements:
  ✓ If statements
  ✓ For loops
  ✓ While loops
  ✓ Labeled statements
  ✓ Switch statements
  ✓ Comments with asterisks
No such file or directory (os error 2)
```
/cc @bfredl

janlazo committed:
vim-patch:8.1.2197: ExitPre autocommand may cause accessing freed memory

Problem:    ExitPre autocommand may cause accessing freed memory.
Solution:   Check the window pointer is still valid. (closes vim/vim#5093)
https://github.com/vim/vim/commit/34ba06b6e6f94bb46062e6c85dbfdcbb0d255ada

janlazo commented:
nvm `--lts` is aliased to v8.12.x. v8.16.x is EOL by end of this year. tree-sitter runs on Node v10.x (LTS) (https://github.com/tree-sitter/tree-sitter/commit/ad43b211f4930404563a3ec65ba1a75e0948c9fc#diff-354f30a63fb0907d4ad57269548329e3R14-R15).

blueyed commented:
See also https://github.com/neovim/neovim/pull/11266 wrt tree-sitter.

janlazo commented on ci/build.ps1:
```diff
@@ -98,27 +98,17 @@ npm.cmd install -g neovim
+Expand-Archive .\tree_sitter_c.zip -DestinationPath .
+cd tree-sitter-c-0.15.2
+tree-sitter.cmd test
+Test-Path -PathType Leaf "$TREE_SITTER_DIR\bin\c.dll"
```
Why is the functional test failing when this test (for the same path) succeeds?
